docker_images/sklearn/tests/test_api_tabular_classification.py

```diff
@@ -38,19 +46,26 @@ def tearDown(self):
         else:
             del os.environ["TASK"]
 
-    def test_simple(self):
+    def _check_requirement(self, requirement):
+        if not requirement:
+            pytest.skip("Skipping test because requirements are not met.")
+
+    def test_success_code(self):
+        # This test does a sanity check on the output and checks the response
+        # code which should be 200. This requires the model to be from the
+        # latest sklearn which is the one installed locally.
+        self._check_requirement(not self.case_data["old_sklearn"])
+
         data = self.data
-        expected_output_len = len(next(iter(data["data"].values())))
+        expected_output_len = len(self.expected_output)
 
         with TestClient(self.app) as client:
             response = client.post("/", json={"inputs": data})
-        self.assertEqual(
-            response.status_code,
-            200,
-        )
+
+        assert response.status_code == 200
         content = json.loads(response.content)
-        self.assertEqual(type(content), list)
-        self.assertEqual(len(content), expected_output_len)
+        assert type(content) == list
```

Comment:
Why are you using asserts instead of `self.assertEqual` for consistency with the rest of this repo's codebase?

Reply:
because we use `pytest` and it correctly understands `assert`. I find `assert x == y` significantly more readable and elegant than `self.assertEqual(x, y)`.

"We should do it the XXX way cause we've done it this way" is not really a good argument ;)

Reply:
Consistency in a codebase is more important to me, especially as other library owners look into other implementations when implementing theirs. Up to you though!

Reply:
They should start using `assert` too :D

Reply:
Actually no, `assert` is not great because we cannot enforce the use `pytest`. `python -m unittest` works just as well.
And if you're not using it, then the error messages are pretty unreadable.

Also `transformers` is also using that format, and pretty much every lib in the HF ecosystem I have seen so far.
It's just the most portable one, so let's go for this one, please.

I won't strongly oppose this as long as the test does its job, but I feel more favorable to `assertEqual`.

Reply:
Note that these tests do use pytest. It's importing it, so w/o `pytest` this PR doesn't work anyway. And we pretty much never ask people to run tests with unittest, we always give them a pytest command to run.

Reply:
If I am not mistaken, you are adding the hard dependency, otherwise the file would be relatively agnostic.

You are free to break consistency here as `sklearn` inference is meant to be isolated from the rest, just commenting about the rest of HF ecosystem.

I'll resolve as I think there's not really anything to add anymore.